Do not update tier prices if getPrice returns undefined values,
Add if statement to prevent JS issues when tierPrices property is missing in priceConfig

diff --git a/view/base/web/js/price-box.js b/view/base/web/js/price-box.js
index 28eaad3..2f4dbce 100644
--- a/view/base/web/js/price-box.js
+++ b/view/base/web/js/price-box.js
@@ -226,6 +226,13 @@ define([
             var originalPrice,
                 prices = {'prices': {}};
 
+            var tierPrice = this.getPrice('price');
+            var tierBasePrice = this.getPrice('basePrice');
+
+            if (!(tierPrice || tierBasePrice)) {
+                return;
+            }
+
             if (this.options.prices.finalPrice) {
                 originalPrice = this.options.prices.finalPrice.amount;
                 prices.prices.finalPrice = {'amount': this.getPrice('price') - originalPrice};
@@ -251,10 +258,12 @@ define([
                 tierPriceItem,
                 i;
 
-            for (i = 0; i < this.options.priceConfig.tierPrices.length; i++) {
-                tierPriceItem = this.options.priceConfig.tierPrices[i];
-                if (productQty >= tierPriceItem.qty && tierPriceItem[priceKey]) {
-                    result = tierPriceItem[priceKey];
+            if (this.options.priceConfig.tierPrices) {
+                for (i = 0; i < this.options.priceConfig.tierPrices.length; i++) {
+                    tierPriceItem = this.options.priceConfig.tierPrices[i];
+                    if (productQty >= tierPriceItem.qty && tierPriceItem[priceKey]) {
+                        result = tierPriceItem[priceKey];
+                    }
                 }
             }
